### PR TITLE
[EM-1735] fix referent news creation

### PR DESCRIPTION
**Pull request by elvino29**

### CODE REVIEW

File: features/api/jecoute_news.feature [hunk 1]
```diff
@@ -506,3 +506,95 @@ Feature:
     }
     """
     Then the response status code should be 403
+
+  Scenario: As a DC referent i cannot create a local news without a zone
```
Comment (illusionOfParadise) [inline features/api/jecoute_news.feature:510]:
```suggestion
  Scenario: As a DC referent I cannot create a local news without a zone
```

File: features/api/jecoute_news.feature [hunk 2]
```diff
@@ -506,3 +506,95 @@ Feature:
...
+    Given I am logged with "referent@en-marche-dev.fr" via OAuth client "Data-Corner"
+    Then I should have 0 notification
```
Comment (illusionOfParadise) [inline features/api/jecoute_news.feature:512]: This check is useless.

File: features/api/jecoute_news.feature [hunk 3]
```diff
@@ -506,3 +506,95 @@ Feature:
...
+        "violations": [
+            {
+                "propertyPath": "zone",
+                "message": "Une zone est nécessaire pour une actualité référente"
+            }
+        ]
+    }
+    """
+
+  Scenario: As a DC referent i cannot create a local news with a city zone
```
Comment (illusionOfParadise) [inline features/api/jecoute_news.feature:540]:
```suggestion
  Scenario: As a DC referent I cannot create a local news with a city zone
```

File: features/api/jecoute_news.feature [hunk 4]
```diff
@@ -506,3 +506,95 @@ Feature:
...
+    Given I am logged with "referent@en-marche-dev.fr" via OAuth client "Data-Corner"
+    Then I should have 0 notification
```
Comment (illusionOfParadise) [inline features/api/jecoute_news.feature:542]: Same here.

File: features/api/jecoute_news.feature [hunk 5]
```diff
@@ -506,3 +506,95 @@ Feature:
...
+      "violations": [
+        {
+          "propertyPath": "zone",
+          "message": "Cette zone ne correspond pas à une région, un département ou un arrondissement"
+        }
+      ]
+    }
+    """
+
+  Scenario: As a DC referent i cannot create a local news with a non managed zone
```
Comment (illusionOfParadise) [inline features/api/jecoute_news.feature:571]:
```suggestion
  Scenario: As a DC referent I cannot create a local news with a non managed zone
```

File: features/api/jecoute_news.feature [hunk 6]
```diff
@@ -506,3 +506,95 @@ Feature:
...
+    Given I am logged with "referent@en-marche-dev.fr" via OAuth client "Data-Corner"
+    Then I should have 0 notification
```
Comment (illusionOfParadise) [inline features/api/jecoute_news.feature:573]: Same here.

**Review by illusionOfParadise — APPROVED** [review]

File: features/api/jecoute_news.feature (hunk 2 above)
Comment (elvino29) [reply features/api/jecoute_news.feature:512]: copy/paste feeling :)

File: src/Jecoute/Api/Listener/PreWriteNewsListener.php [hunk 7]
```diff
@@ -3,33 +3,26 @@
...
-    private Security $security;
     private NewsHandler $handler;
     private RequestStack $requestStack;
     private AuthorizationChecker $authorizationChecker;
 
     public function __construct(
-        Security $security,
         NewsHandler $handler,
         RequestStack $requestStack,
         AuthorizationChecker $authorizationChecker
```
Comment (ottaviano) [inline src/Jecoute/Api/Listener/PreWriteNewsListener.php:24]: could you remove these services as unused

File: src/Normalizer/JecouteNewsNormalizer.php [hunk 8]
```diff
@@ -61,6 +72,11 @@ public function denormalize($data, $type, $format = null, array $context = [])
         /** @var News $news */
         $news = $this->denormalizer->denormalize($data, $type, $format, $context);
 
+        $scope = $this->authorizationChecker->getScope($this->requestStack->getMasterRequest());
+        if (ScopeEnum::REFERENT === $scope) {
+            $news->setSpace(JecouteSpaceEnum::REFERENT_SPACE);
+        }
+
```
Comment (ottaviano) [inline src/Normalizer/JecouteNewsNormalizer.php:79]: we should do this only on creation flow (POST request)

**Review by ottaviano — APPROVED** [review]

**Pull request merged**